internal/identity/errors.go

```diff
@@ -0,0 +1,16 @@
+package identity
+
+import "errors"
+
+// ErrRevokeNotImplemented error type when Revoke method is not implemented
+// by an identity provider
+var ErrRevokeNotImplemented = errors.New("revoke not implemented")
+
+// ProviderError records an error and the message and provider involved.
+type ProviderError struct {
+	Provider string
+	Source   string
+	Err      error
+}
+
+func (e ProviderError) Error() string { return e.Source + " " + e.Provider + ": " + e.Err.Error() }
```

Comment:
What if e.Err is nil?

Reply:
If the error is nil, then we'll have a nil pointer error, a segmentation violation runtime error, I honestly didn't think anyone would want to use the `ProviderError` without actually wrapping an error in there.

To avoid this, I can do it this way 
`func (e ProviderError) Error() string { return fmt.Sprintf("%s %s: %v", e.Source, e.Provider, e.Err) }`.

What do you think?